src/scenes/home/signup/signup.js

```diff
@@ -134,7 +138,7 @@ class SignUp extends Component {
             <li className={styles.errorMessage}>{this.state.error}</li>
           </ul> : null }
           {this.state.success && <Redirect to="/thanks" />}
-          <FormButton className={styles.joinButton} text="Join" onClick={this.handleOnClick} theme="red" />
+          {this.state.isLoading ? <FormButton className={styles.joinButton} text="Loading..." onClick={this.handleOnClick} theme="red" /> : <FormButton className={styles.joinButton} text="Join" onClick={this.handleOnClick} theme="red" />}
```

Comment:
Can you also make the button disabled once it's submitted? That should help cut down on multiple requests being fired.